app/main.py

```diff
@@ -22,161 +23,163 @@
-			flash(filename.replace('.','-'),'success-zip')
-		elif not _zip and _pdf:
-			flash(filename.replace('.','-'),'success-pdf')
-
-		return redirect(url_for('index'))
+    empty_directory()
+    csv = request.form['csv'].strip()
+    img = request.form['img-default']
+    eventyay_url = request.form['eventyay_url'].strip()
+    text_on_image = request.form['text_on_image']
+    file = request.files['file']
+    _pdf = True if request.form.get('pdf') == 'on' else False
+    _zip = True if request.form.get('zip') == 'on' else False
+
+    if not _pdf and not _zip:
+        flash('Please select a download filetype!', 'error')
+        return redirect(url_for('index'))
+
+    # If default background is selected
+    if img != '':
+        if(img == 'user_defined.png'):
+            bg_color = request.form['bg_color']
+            text_on_image = request.form['text_on_image']
+            do_svg2png(img, 1, bg_color, text_on_image)
+            filename = img + '.csv'
+
+        filename = img + '.csv'
+    # If the textbox is filled
+    elif csv != '':
+        check_csv = csv.splitlines()
+        count_line = 0
+        for check in check_csv:
+            line = check.split(',')
+            if len(line) == 4:
+                count_line = count_line + 1
+        if count_line == len(check_csv):
+            filename = "default.png.csv"
+            f = open(os.path.join(app.config['UPLOAD_FOLDER'], filename), "w+")
+            f.write(csv)
+            f.close()
+        else:
+            flash('Write Data in Correct format!', 'error')
+            return redirect(url_for('index'))
+    # if user does not select file, browser submits an empty part without filename
+    elif eventyay_url != '':
+        filename = 'speaker.png.csv'
+        generate_csv_eventyay.tocsv(eventyay_url, filename)
+    else:
+        if file.filename == '':
+            flash('Please select a CSV file to Upload!', 'error')
+            return redirect(url_for('index'))
+        else:
+            filename = str(uuid.uuid4()) + '.png.csv'
+
+    # If a PNG is uploaded, push it to the folder
+    if request.files['image'].filename != '':
+        image = request.files['image']
+        imgname = image.filename
+        if '.' in imgname and imgname.rsplit('.', 1)[1] == 'png':
+            imgname = filename.rsplit('.', 1)[0]
+            imgname = secure_filename(imgname)
+            image.save(os.path.join(app.config['UPLOAD_FOLDER'], imgname))
+        else:
+            flash('Please select a PNG image to Upload!', 'error')
+            return redirect(url_for('index'))
+    else:
+        if filename.find("png.csv") != -1:
+            if img == '':
+                flash('Please upload an image in \'PNG\' format!', 'error')
+                return redirect(url_for('index'))
+        else:
+            flash('Please upload a file in \'CSV\' format! Each line is a badge and the lines on the badge are separated by commas. Try it out!', 'error')
```

Comment:
@bhaveshAn What about this?

Reply:
@vaibhavsingh97 since this is a single function on one line, hence can't be split into two lines.
It would become unlogical if the `flake8` rules are followed here. Hence I would like to retain it.

Reply:
btw we can use \ to  split this lines into two and functionality won't be compramised 😅